test/extension-tests/successful-build/test/variable-substitution.test.ts

```diff
@@ -0,0 +1,184 @@
+    expect(testEnv.projectFolder.buildDirectory.isCMakeCachePresent).to.eql(true, 'expected cache not present');
+    const cache = await CMakeCache.fromPath(await cmt.cachePath);
+
+    const cacheEntry = cache.get('generator') as api.CacheEntry;
+    expect(cacheEntry.type).to.eq(api.CacheEntryType.String, '[generator] unexpected cache entry type');
+    expect(cacheEntry.key).to.eq('generator', '[generator] unexpected cache entry key name');
+    const generator = cache.get('CMAKE_GENERATOR') as api.CacheEntry;
+    expect(cacheEntry.as<string>()).to.eq(generator.as<string>(), '[generator] substitution incorrect');
+    expect(typeof cacheEntry.value).to.eq('string', '[generator] unexpected cache entry value type');
+  }).timeout(60000);
+
+  test('Check substitution for "projectName"', async () => {
+    // Set fake settings
+    testEnv.setting.changeSetting('configureSettings', {projectName: '${projectName}'});
+
+    // Configure
+    expect(await cmt.configure()).to.be.eq(0, '[projectName] configure failed');
+    expect(testEnv.projectFolder.buildDirectory.isCMakeCachePresent).to.eql(true, 'expected cache not present');
+    const cache = await CMakeCache.fromPath(await cmt.cachePath);
+
+    const cacheEntry = cache.get('projectName') as api.CacheEntry;
+    expect(cacheEntry.type).to.eq(api.CacheEntryType.String, '[projectName] unexpected cache entry type');
+    expect(cacheEntry.key).to.eq('projectName', '[projectName] unexpected cache entry key name');
+    expect(cacheEntry.as<string>()).to.eq('Unknown Project', '[projectName] substitution incorrect');
+    expect(typeof cacheEntry.value).to.eq('string', '[projectName] unexpected cache entry value type');
+  }).timeout(60000);
+
+  test('Check substitution for "userHome"', async () => {
+    // Set fake settings
+    testEnv.setting.changeSetting('configureSettings', {userHome: '${userHome}'});
+
+    // Configure
+    expect(await cmt.configure()).to.be.eq(0, '[userHome] configure failed');
+    expect(testEnv.projectFolder.buildDirectory.isCMakeCachePresent).to.eql(true, 'expected cache not present');
+    const cache = await CMakeCache.fromPath(await cmt.cachePath);
+
+    const cacheEntry = cache.get('userHome') as api.CacheEntry;
+    expect(cacheEntry.type).to.eq(api.CacheEntryType.String, '[userHome] unexpected cache entry type');
+    expect(cacheEntry.key).to.eq('userHome', '[userHome] unexpected cache entry key name');
+    const user_dir = process.platform === 'win32' ? process.env['HOMEPATH']! : process.env['HOME']!;
+    expect(cacheEntry.as<string>()).to.eq(user_dir, '[userHome] substitution incorrect');
+    expect(typeof cacheEntry.value).to.eq('string', '[userHome] unexpected cache entry value type');
+  }).timeout(60000);
+
+  test('Check substitution for variant names', async () => {
+    // Set fake settings
+    testEnv.setting.changeSetting('configureSettings', {buildLabel: '${buildLabel}', otherVariant: '${otherVariant}'});
```

Comment:
Variable Substitution has no describtion for '${buildLabel}' and '${otherVariant}'.

Reply:
@KoeMai `${buildLabel}` is for the `buildType` variant option. It is available for substitution via `buildLabel`. `${otherVariant}` is the name of a variant option I defined in the test project folder.

Reply:
I think we need in the documentation a description for the replacement. Do I understand this right?
The buildLabel and otherVariant are the identifiers for the variants. The ${idenfier} is replaced on configuration by the selected choice.

Reply:
Some users requested for the variant names to be available for variable substitution, which is why I added it.

I have defined the following variant config in the test project folder:
```json
{
    "buildType": {
        "default": "debug-label",
        "choices": {
            "debug-label": {
                "short": "debug-label short",
                "buildType": "Debug"
            },
            "not-debug": {
                "short": "not-debug short",
                "buildType": "Release"
            }
        }
    },
    "otherVariant": {
        "default": "option1",
        "choices": {
            "option1": {
                "short": "option1 short"
            },
            "option2": {
                "short": "option2 short"
            }
        }
    }
}
```

And it is pretty much how you explained it.
I had to replace the `buildType` variant with another name to avoid name collisions.

Reply:
I have added some documentation about the variant identifier substitution.

Reply:
Where's the doc change? I don't see it.

Reply:
@vector-of-bool I added them in the docs files `docs/configuring.rst`